Admin/Test-AMSI.ps1

```diff
@@ -1,144 +1,793 @@
+                        Write-Host "Refreshed Get-ExchangeDiagnosticInfo on $server"
+                    }
+                    $needsRefresh++
+                } else {
+                    Write-Warning "AMSI is already disabled on Exchange configuration by SettingOverride $($getSO.Identity)"
+                }
+            }
+        }
+
+        if ( $needsRefresh -gt 0 -and ($SupportedExchangeServers.Site.Name | Get-Unique).count -gt 1) {
+            Write-Host ""
+            Write-Host $bar
+            Write-Host ""
+            Write-Warning "You have a multi site environment, confirm that all affected Exchange sites has replicated changes."
+            Write-Host "You can push changes on your DCs with: repadmin /syncall /AdeP"
+            Write-Host ""
+            Write-Warning "Remember to restart IIS to be effective."
+            Write-Host "You can accomplish this by running .\Test-AMSI.ps1 -RestartIIS"
+            Write-Host ""
+        }
+
+        if ( $RestartIIS ) {
+            if ($filterList) {
+                $yesToAll = $false
+                $noToAll = $false
+
+                if ( $Force -or $PSCmdlet.ShouldContinue("Are you sure you want to do it?", "This command wil restart the following IIS servers: $filterList", $true, [ref]$yesToAll, [ref]$noToAll) ) {
+                    foreach ($server in $filterList) {
+                        Write-Host $msgNewLine
+                        if ( $Force -or $filterList.Count -eq 1 -or $PSCmdlet.ShouldContinue("Are you sure you want to do it?", "You will restart IIS on server $server", $true, [ref]$yesToAll, [ref]$noToAll) ) {
+                            Get-ExchangeDiagnosticInfo -Process Microsoft.Exchange.Directory.TopologyService -Component VariantConfiguration -Argument Refresh -Server $server | Out-Null
+                            Write-Host "Refreshed Get-ExchangeDiagnosticInfo on $server"
+                            if ( $server.ToLower() -eq $env:COMPUTERNAME.ToLower() ) {
+                                Write-Host "Restarting local IIS on $server"
+                                Get-Service W3SVC, WAS | Restart-Service -Force
+                            } else {
+                                Write-Host "Restarting Remote IIS on $server"
+                                Get-Service W3SVC, WAS -ComputerName $server | Restart-Service -Force
+                            }
+                            Write-Host "$server Restarted"
+                        }
+                    }
+                }
+            }
+        }
+    } else {
+        Write-Warning "We did not find Exchange servers with AMSI support"
+    }
+    Write-Host $bar
+    Write-Host ""
+    Write-Host 'You can find additional information at:'
+    Write-Host 'https://techcommunity.microsoft.com/t5/exchange-team-blog/more-about-amsi-integration-with-exchange-server/ba-p/2572371' -ForegroundColor Cyan
```

Comment:
It might be better to create an aka.ms short link for this (not sure if we already have one for the AMSI blog post)?

Reply:
That's fine with me.
Do you want me to do it or do you do it since you have ownership of the previous ones?

Reply:
@tweekerz @ninobb do you know if we already have a short-link for `https://techcommunity.microsoft.com/t5/exchange-team-blog/more-about-amsi-integration-with-exchange-server/ba-p/2572371` ?

Reply:
@iserrano76 it doesn't look like we have one and so, I've created one: https://aka.ms/ExchangeAMSI